packages-next/keystone/src/session/index.ts

```diff
@@ -88,17 +88,27 @@ export function withItemData(createSession: CreateSession, fieldSelections: Fiel
         ) {
           return;
         }
-        // If no field selection is specified, just load the id. We still load the item,
-        // because doing so validates that it exists in the database
-        const item = await sudoContext.lists[session.listKey].findOne({
-          where: { id: session.itemId },
-          resolveFields: fieldSelections[session.listKey] || 'id',
-        });
-        // If there is no matching item found, return no session
-        if (!item) {
+
+        // NOTE: This is wrapped in a try-catch block because a "not found" result will currently
+        // throw; I think this needs to be reviewed, but for now this prevents a system crash when
+        // the session item is invalid
+        try {
+          // If no field selection is specified, just load the id. We still load the item,
+          // because doing so validates that it exists in the database
+          const item = await sudoContext.lists[session.listKey].findOne({
+            where: { id: session.itemId },
+            resolveFields: fieldSelections[session.listKey] || 'id',
+          });
+          // If there is no matching item found, return no session
+          if (!item) {
+            return;
+          }
```

Comment:
```suggestion
```

Is there ever an instance where `item` is falsey?

Reply:
There **was** - tbh I'm not confident enough about the behaviour here to want to remove this code path. It's super defensive, but anything unhandled here completely blocks the request so for now I'd prefer to land this with a big TODO around it and then review it more thoroughly

Reply:
OK, I'm pretty confident that this code path is impossible, but not going to block it 👍